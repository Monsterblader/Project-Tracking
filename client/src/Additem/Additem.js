import React from 'react';
import Plus from '../assets/Plus.js';
import './Additem.css';

function Additem(props) {
  return (
    <div className="add-item">
      <div
          className="svg-container"
          onClick={ e => {
            props.activateAdditem(e.currentTarget.nextSibling.value);
            e.currentTarget.nextSibling.value = '';
          }}>
        <Plus />
      </div>
      <input
        type="text"
        className="text-box"
        placeholder="Another day, another task"
        style={{ width: '200px' }}
      />
    </div>
  );
}

export default Additem;
