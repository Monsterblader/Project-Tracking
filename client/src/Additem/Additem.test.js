import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import Renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import Additem from './Additem.js';

jest.mock('../assets/Plus.js',
  () =>
    () => <div id="mock-Plus">Plus</div>
);

describe('Additem', () => {
  let container = null;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test('it renders "Additem"', () => {
    const component = Renderer.create(
      <Additem />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    render(<Additem />, container);
  });

  describe('the <Plus />', () => {
    it('renders <Plus />', () => {
      render(<Additem />, container);
      expect(document.getElementById('mock-Plus')).toBeTruthy();
    });

    it('calls updateItem on click', () => {
      const handleSubmit = Additem.prototype.handleSubmit;
      const testSubmit = jest.fn();
      Additem.prototype.handleSubmit = testSubmit;
      render(<Additem />, container);
      const plus = document.getElementById('mock-Plus');

      plus.parentNode.click();
      expect(testSubmit).toHaveBeenCalledTimes(1);

      Additem.prototype.handleSubmit = handleSubmit;
    });
  });

  describe('handleSubmit', () => {
    it('should call updateItem and cancel', () => {
      const activateAdditem = jest.fn();
      const context = {
        props: { activateAdditem },
        input: { current: { value: 'new value' } },
      }
      const e = { preventDefault: jest.fn() };
      const handleSubmit = Additem.prototype.handleSubmit.bind(context);

      handleSubmit(e);
      expect(activateAdditem).toHaveBeenCalledWith('new value');
    });

    it('handles submit', () => {
      const handleSubmit = Additem.prototype.handleSubmit;
      const testSubmit = jest.fn();
      Additem.prototype.handleSubmit = testSubmit;

      render(<Additem />, container);
      const el = document.getElementsByTagName('input')[0];
      // const form = document.getElementsByTagName('form')[0];

      // ReactTestUtils.Simulate.change(el);
      // ReactTestUtils.Simulate.keyDown(form, { key: 'Enter', keyCode: 13, which: 13 });
      ReactTestUtils.Simulate.submit(el);
      expect(testSubmit).toHaveBeenCalledTimes(1);

      Additem.prototype.handleSubmit = handleSubmit;
    });
  });
});
