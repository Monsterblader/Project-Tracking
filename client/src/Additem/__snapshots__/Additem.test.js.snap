// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Additem it renders "Additem" 1`] = `
<form
  className="add-item"
  onSubmit={[Function]}
>
  <div
    className="svg-container"
    onClick={[Function]}
  >
    <div
      id="mock-Plus"
    >
      Plus
    </div>
  </div>
  <input
    className="text-box"
    placeholder="Another day, another task"
    type="text"
  />
</form>
`;
